dear_petition/petition/types/dismissed.py

```diff
@@ -28,3 +28,35 @@ def build_guilty_to_lesser_query():
     plea = Q(offense__plea="GUILTY TO LESSER")
     query = action & method & plea
     return query
+
+
+def same_day_convictions(dismissed_records):
+    """
+    To support field 3(b) on the AOC-CR-287 form, return a list of guilty
+    OffenseRecords that also had dismissals on the same day.
+
+    Petition language:
+        I was charged with multiple offenses, and while the charges
+        listed above were dismissed, the following charges resulted in a
+        conviction on the day of the dismissal or had not yet reached final
+        disposition.
+    """
+    dismissed_record_ids = dismissed_records.values_list("id", flat=True)
+    qs = (
+        OffenseRecord.objects.filter(pk__in=list(dismissed_record_ids))
```

Comment:
I actually had that exact query before 9043bcaf7a6b474b65f55477f7104be7bc53f065. IIRC, since `dismissed_records` is sourced from `get_all_offense_records()`, which itself is a complex query, `distinct()` wasn't working properly. It's been a few months since I wrote it though... so I don't fully recall.